File: ScalaBaseline/src/aeminium/actors/baseline/BenchmarkActor.scala
package aeminium.actors.baseline

import scala.actors.Actor
import aeminium.actormodel.examples.dictionary.BenchmarkBackend

case object Stop
case class GetValue(val key:Int)
case class SetValue(val key:Int, val obj:Int)

class BenchmarkActor() extends Actor {
	val backend:BenchmarkBackend = new BenchmarkBackend(BenchmarkActor.messages);
	
	def act() {
		loop {
			react {
				case GetValue(i) => backend.getValue(i)
				case SetValue(i,o) => backend.setValue(i,o)
				case Stop =>
					sender ! Stop
					exit()
			}
		}
	}
	
}

object BenchmarkActor {
	val messages = 1000;
	
	def main(args: Array[String]) {
		val startTime = System.nanoTime()
		
		val ba = new BenchmarkActor
		ba.start
		1.to(messages).foreach { i =>
			if (i % 5 == 0) {
				ba ! SetValue(i,10)
			} else {
				ba ! GetValue(i)
			}
		}
		ba !? Stop
		java.lang.System.out.println("Took (nanoseconds): " + (System.nanoTime() - startTime))
	}
}

File: AeminiumAM/src/aeminium/actormodel/examples/dictionary/BenchmarkActor.java
package aeminium.actormodel.examples.dictionary;

import aeminium.actormodel.actor.Actor;
import aeminium.actormodel.annotations.Read;
import aeminium.actormodel.annotations.Write;

public class BenchmarkActor extends Actor {
	
	protected final static int expectedMessages = 1000;
	private final static BenchmarkBackend backend = new BenchmarkBackend(23);
	
	@Write
	public void setValue(int key, int value) {
		backend.setValue(key, value);
	}
	
	@Read
	public void getValue(int key) {
		backend.getValue(key);
	}
	
	public static void main(String[] args) {
		BenchmarkActor ba = new BenchmarkActor();
		
		for (int i = 0; i < expectedMessages; i++) {
			if (i % 5 == 0) {
				ba.setValue(i, 10);
			} else {
				ba.getValue(i);
			}
			
		}
		
	}
	
}
